core/src/test/java/org/mule/tck/ZipUtils.java

```diff
@@ -0,0 +1,108 @@
+/*
+ * Copyright (c) MuleSoft, Inc.  All rights reserved.  http://www.mulesoft.com
+ * The software in this package is published under the terms of the CPAL v1.0
+ * license, a copy of which has been included with this distribution in the
+ * LICENSE.txt file.
+ */
+
+package org.mule.tck;
+
+import org.mule.util.StringUtils;
+
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.util.zip.ZipEntry;
+import java.util.zip.ZipOutputStream;
+
+/**
+ * Provides utility methods to work with ZIP files
+ */
+public class ZipUtils
+{
+
+    private ZipUtils()
+    {
+    }
+
+    /**
+     * Describes a resource that can be compressed in a ZIP file
+     */
+    public static class ZipResource
+    {
+
+        private final File file;
+        private final String alias;
+
+        public ZipResource(File file)
+        {
+            this(file, null);
+        }
+
+        public ZipResource(File file, String alias)
+        {
+            this.file = file;
+            this.alias = alias;
+        }
+
+        public String getName()
+        {
+            return StringUtils.isEmpty(alias) ? file.getName() : alias;
+        }
+
+        public File getFile()
+        {
+            return file;
+        }
+    }
+
+    /**
+     * Compress a set of resource files into a ZIP file
+     *
+     * @param targetFile file that will contain the zipped files
+     * @param resources resources to compress
+     * @throws IOException in case of any error processing the files
+     */
+    public static void compress(File targetFile, ZipResource[] resources) throws IOException
+    {
+        ZipOutputStream out = new ZipOutputStream(new FileOutputStream(targetFile));
+
+        try
+        {
+            for (ZipResource zipResource : resources)
+            {
+
+                FileInputStream in = new FileInputStream(zipResource.getFile());
+                try
+                {
+                    out.putNextEntry(new ZipEntry(zipResource.getName()));
+
+                    byte[] b = new byte[1024];
+
+                    int count;
+
+                    while ((count = in.read(b)) > 0) {
```

Comment:
This is reading/writing a chunk of bytes. I'll rename the variable to make it more clear